### ISSUE TITLE
error: expected primary-expression before ',' token

### ISSUE THREAD

**Issue by jupiterjosh** [mattreecebentley/plf_colony#16]
Hi Matt, 

Colony looks great.  I am very excited to be using this.  It's been about 15 years since I last developed in C++ and I came across your website (and your Gamasutra articles) while I was researching what is new in C++ since 2002.  I am developing a game using the Godot engine and colony is the exact kind of container I'm looking for.

I am having difficulties getting colony to compile with either clang or gcc on a linux box.  It appears that the compilers are having issues with an optional memset optimization in the blank() function.  The gcc output is slightlly different and I have included both below.  As a workaround I commented out the optional memset block (lines 1054-1062) to get it to compile.  From what I could tell commenting out the code shouldn't impact the proper working of colony or materially impact its performance (at least I hope this is the case; my C++ is rusty).

I'm fairly sure this isn't a critical error/issue, but I wanted to make you aware of it all the same.

Thanks for your efforts,

Joshua

Error with clang
```
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
clang++ -o src/common.os -c -g -O3 -std=c++11 -Wno-writable-strings -fPIC -I/home/joshua/Desktop/sdf_local/godot_headers -I/home/joshua/Desktop/sdf_local/godot-cpp/include -I/home/joshua/Desktop/sdf_local/godot-cpp/include/core -Isrc src/common.cpp
In file included from src/common.cpp:20:
In file included from src/csbx_force_sim.hpp:9:
src/plf_colony.h:1058:61: error: 'colony' does not refer to a value
  ...std::memset(reinterpret_cast<void *>(this), 0, offsetof(colony, pointer_allocator_pair));
                                                             ^
src/plf_colony.h:209:149: note: declared here
  ...typename element_skipfield_type = unsigned short > class colony : private element_allocator_...
                                                              ^
1 error generated.
scons: *** [src/common.os] Error 1
scons: building terminated because of errors.

```
Error with gcc
```
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
g++ -o src/common.os -c -g -O3 -std=c++11 -Wno-writable-strings -fPIC -I/home/joshua/Desktop/sdf_local/godot_headers -I/home/joshua/Desktop/sdf_local/godot-cpp/include -I/home/joshua/Desktop/sdf_local/godot-cpp/include/core -Isrc src/common.cpp
In file included from src/csbx_force_sim.hpp:9:0,
                 from src/common.cpp:20:
src/plf_colony.h: In member function 'void plf::colony<element_type, element_allocator_type, element_skipfield_type>::blank()':
src/plf_colony.h:1058:67: error: expected primary-expression before ',' token
     std::memset(reinterpret_cast<void *>(this), 0, offsetof(colony, pointer_allocator_pair));
```

**Comment by mattreecebentley** [comment]
I'm just uploading v5.0 now, check that out and tell me whether the same error exists -

**Comment by jupiterjosh** [comment]
Will do.  Thanks.

**Comment by mattreecebentley** [comment]
BTW I tested on gcc 7.3 & clang 6 in xubuntu and couldn't find anything, so if you spot something let me know the compiler version etc.
Thank you,

**Comment by mattreecebentley** [comment]
Nevermind, I know what is causing this - will be fixed in the next upload, within a few days- thanks

**Comment by mattreecebentley** [comment]
Fixed via v5.01 - thanks for spotting this,.
